src/applications/vaos/services/appointment/transformers.js

```diff
@@ -233,9 +233,21 @@ function getPurposeOfVisit(appt) {
 function getRequestedPeriods(appt) {
   const requestedPeriods = [];
   const format = 'MM/DD/YYYY';
+
   for (let x = 1; x <= 3; x += 1) {
     const optionTime = appt[`optionTime${x}`];
-    if (optionTime) {
+
+    if (!moment(appt[`optionDate${x}`]).isValid()) {
+      // Since 'No Date Selected' and 'No Time Selected' are invalid dates and times,
+      // don't add the start and end attributes.
+      //
+      // NOTE: FHIR Spec says...
+      // If the start element is missing, the start of the period is not known.
+      // If the end element is missing, it means that the period is ongoing,
+      // or the start may be in the past, and the end date in the future,
+      // which means that period is expected / planned to end at the specified time.
+      requestedPeriods.push({});
```

Comment:
I think we could skip adding anything to `requestedPeriods` in the case where the date/time isn't selected.